dashboard/src/components/AppView/AppControls.tsx

```diff
@@ -28,19 +28,23 @@ class AppControls extends React.Component<IAppControlsProps, IAppControlsState>
 
   public render() {
     const { name, namespace } = this.props.app;
+    const deleted = this.props.app.info && this.props.app.info.deleted;
     if (!name || !namespace) {
       return <div> Loading </div>;
     }
     return (
       <div className="AppControls">
-        <button className="button" onClick={this.handleUpgradeClick}>
-          Upgrade
-        </button>
+        {/* If the app has been deleted hide the upgrade button */}
+        {!deleted && (
+          <button className="button" onClick={this.handleUpgradeClick}>
+            Upgrade
+          </button>
+        )}
         {this.state.upgrade && (
           <Redirect push={true} to={`/apps/ns/${namespace}/upgrade/${name}`} />
         )}
         <button className="button button-danger" onClick={this.openModel}>
-          Delete
+          {deleted ? "Purge" : "Delete"}
```

Comment:
I think this is fine, since in #481 we'll need this button to say "Delete" for running apps, anyway. Plus, changing the button to always be "Purge" in this PR will change the current UX, when we only want to change this for deleted apps in this PR.

Reply:
yes, I agree with @prydonius. I think that using `Purge` always is unnecessary verbose (and a bit confusing if a user see this changing).